.github/workflows/ci-publish.yml

```diff
@@ -84,6 +87,5 @@ jobs:
           TWINE_USERNAME: ${{ secrets.PYPI_USERNAME }}
           TWINE_PASSWORD: ${{ secrets.PYPI_PASSWORD }}
         run: |
-          python -m pip install --upgrade pip
-          pip install twine
-          twine upload dist/*
+          poetry build
```

Comment:
Suggest removing this build command, as the package should already been built in a previous step. It may be worth checking that poetry uses the same `dist` directory for the python wheels.